roles/kubernetes/preinstall/tasks/resolvconf.yml

```diff
@@ -30,21 +21,32 @@
     marker: "# Ansible entries {mark}"
   notify: Preinstall | restart network
 
-- name: Add options to resolv.conf
-  lineinfile:
-    line: options {{ item }}
-    dest: "{{resolvconffile}}"
-    state: present
-    regexp: "^options.*{{ item }}$"
-    insertafter: EOF
+- name: Remove search/domain/nameserver options before block
+  replace:
+    dest: "{{item[0]}}"
+    regexp: '^{{ item[1] }}[^#]*(?=# Ansible entries BEGIN)'
     backup: yes
     follow: yes
-  with_items:
-    - ndots:{{ ndots }}
-    - timeout:2
-    - attempts:2
+  with_nested:
+    - "{{ [resolvconffile] + [base|default('')] + [head|default('')] }}"
+    - [ 'search ', 'nameserver ', 'domain ', 'options ' ]
+  when: item[0] != ""
   notify: Preinstall | restart network
 
+- name: Remove search/domain/nameserver options after block
+  replace:
+    dest: "{{item[0]}}"
+    regexp: '(# Ansible entries END\n(?:(?!^{{ item[1] }}).*\n)*)(?:^{{ item[1] }}.*\n?)+'
+    replace: '\1'
+    backup: yes
+    follow: yes
+  with_nested:
+    - "{{ [resolvconffile] + [base|default('')] + [head|default('')] }}"
```

Comment:
you may use map extract patter to have more understandable format

`- "{{ [ resolvconffile, base|default(''), head|default('') ] | difference(['']) }}"
`
in this case you will not need to have 'when: item[0] != ""' condition